### ISSUE TITLE
[enhancement]: opentofu support

### ISSUE THREAD

**Issue by Alexlofern** [microsoft/azure-pipelines-agent#5044]
### Describe your feature request here

OpenTofu is a fork of Terraform that is open-source, community-driven, and managed by the Linux Foundation.

Please create an option to switch from terraform to openTofu

https://opentofu.org/

**Comment by KonstantinTyukalov** [comment]
Hi @Alexlofern could you please describe your request in more detail? What type of support is expected?

**Comment by Alexlofern** [comment]
Yes sure, OpenTofu is an open-source Infrastructure-as-Code (IaC) tool, originally a fork of Terraform. It was created to maintain compatibility with existing Terraform configurations while fostering a more open and community-driven development process. OpenTofu enables users to define and provision infrastructure using code in a declarative format, supporting multiple cloud providers and services.

This tool has gained popularity due to changes in Terraform's licensing model, which have led some users to explore alternatives. OpenTofu, being open-source, provides a compelling option for those seeking a fully community-driven and transparent solution without licensing restrictions.

For your Azure DevOps extension functionality, you could propose adding a configuration setting or toggle to allow users to switch seamlessly between Terraform and OpenTofu binaries. This feature would address the licensing concerns and give users the flexibility to adopt OpenTofu while maintaining compatibility with existing workflows and project requirements.
